sdk/schemaregistry/azure-schemaregistry-avroencoder/azure/schemaregistry/encoder/avroencoder/_utils.py

```diff
@@ -0,0 +1,12 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+
+def get_http_request_kwargs(kwargs):
+    http_request_keywords = ["params", "headers", "json", "data", "files"]
+    http_request_kwargs = {
+        key: kwargs.pop(key, None) for key in http_request_keywords if key in kwargs
+    }
+    return http_request_kwargs
```

Comment:
yes, these are the only ones that we filter out. I took this from the `azure-schemaregistry` code, because we had this problem there as well. And there, I traced it back to what the `core` method would take.

Reply:
I looked into the code of `azure-schemaregirtry`:

- get_http_request_kwargs is to extract the arguments used by building a `request`
- other kwargs are kept for `send_request` method

```python
    def get_schema(self, schema_id, **kwargs):
        # type: (str, Any) -> Schema
        http_request_kwargs = get_http_request_kwargs(kwargs)
        request = schema_rest.build_get_by_id_request(
            id=schema_id, **http_request_kwargs
        )
        response = self._generated_client.send_request(request, **kwargs)
        response.raise_for_status()
        return _parse_response_schema(response)
```

I would you not need to duplicate this filter process in the avro encoder since it's taken care by `azure-schemaregistry` already, besides it's dropping the kwargs for `send_request`.

or did you encounter a real error not doing such?

Reply:
ah good catch, this is a bug here: https://github.com/Azure/azure-sdk-for-python/blob/f363c224e41fc2878ff595901451c0497941b4c7/sdk/schemaregistry/azure-schemaregistry/azure/schemaregistry/_schema_registry_client.py#L141

and send_request takes the following kwargs:
`stream`: https://github.com/Azure/azure-sdk-for-python/blob/f363c224e41fc2878ff595901451c0497941b4c7/sdk/core/azure-core/azure/core/_pipeline_client.py#L189
`_return_pipeline_response`: https://github.com/Azure/azure-sdk-for-python/blob/f363c224e41fc2878ff595901451c0497941b4c7/sdk/core/azure-core/azure/core/_pipeline_client.py#L194
`transport`/`policies`/`per_call_policies`/`per_retry_policies`: https://github.com/Azure/azure-sdk-for-python/blob/f363c224e41fc2878ff595901451c0497941b4c7/sdk/core/azure-core/azure/core/_pipeline_client.py#L114
and I think a bunch more from inside `_build_pipeline` method.

Reply:
`request_id`/`auto_request_id`/`response_encoding`/`network_span_namer`/`tracing_attributes`/`session`/`session_owner`/`use_env_settings` are all things here. I'm not sure if we document these as reserved words anywhere.

Reply:
OOB: not passing kwargs when calling schema_registry_client methods for now